lib/rollbar/plugins/basic_socket.rb

```diff
@@ -1,16 +1,29 @@
 Rollbar.plugins.define('basic_socket') do
+  load_on_demand
+
   dependency { !configuration.disable_core_monkey_patch }
 
   # Needed to avoid active_support (< 4.1.0) bug serializing JSONs
-  dependency { defined?(ActiveSupport::VERSION::STRING) }
+  dependency do
+    defined?(ActiveSupport::VERSION::STRING) &&
+      Gem::Version.new(ActiveSupport::VERSION::STRING) < Gem::Version.new('5.2.0')
+  end
+
+  @original_as_json = ::BasicSocket.public_instance_method(:as_json)
 
   execute do
     require 'socket'
 
-    class BasicSocket
-      def as_json(*)
-        to_s
+    class BasicSocket # :nodoc:
+      def as_json(_options = nil)
+        {
+          :value => inspect
+        }.to_json
```

Comment:
While I think the original `to_s` was fine, `#as_json` is actually expected to return a hash. So you could either drop the `.to_json`, or just return `inspect` or `to_s`. If you're going to return a string anyway, might as well make it simple.

Reply:
Great catch. Thank you. I'd rather stick to what `#as_json` is supposed to return instead of returning a string.